packages/cart/src/models/cart.ts

```diff
@@ -1,10 +1,119 @@
+import { DAL } from "@medusajs/types"
 import { generateEntityId } from "@medusajs/utils"
-import { BeforeCreate, Entity, OnInit, PrimaryKey } from "@mikro-orm/core"
+import {
+  BeforeCreate,
+  Collection,
+  Entity,
+  OnInit,
+  OneToMany,
+  OneToOne,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import Address from "./address"
+import LineItem from "./line-item"
+import ShippingMethod from "./shipping-method"
 
-@Entity()
+type OptionalCartProps =
+  | "shipping_address"
+  | "billing_address"
+  | DAL.EntityDateColumns // TODO: To be revisited when more clear
+
+@Entity({ tableName: "cart" })
 export default class Cart {
   @PrimaryKey({ columnType: "text" })
-  id!: string
+  id: string
+
+  @Property({ columnType: "text", nullable: true })
+  region_id?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  customer_id?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  sales_channel_id?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  email?: string | null
+
+  @Property({ columnType: "text" })
+  currency_code: string
+
+  @OneToOne({
+    entity: () => Address,
+    joinColumn: "shipping_address_id",
+    orphanRemoval: true,
+  })
+  shipping_address?: Address | null
+
+  @OneToOne({
+    entity: () => Address,
+    joinColumn: "billing_address_id",
+    orphanRemoval: true,
```

Comment:
> are we sure we want orphan removal and not cascade remove?

Orphan removal might be too aggressive. I thought it would make sense to avoid having unused addresses in the `cart_address` table, but for this, cascading should be enough.

Reply:
I think it should be enough, I am not entirely sure about the orphan removal behaviour. it seems it can also impact the way we manage a collection before persisting it